### PR TITLE
fix(server+clent): added validitions in both fe and be

### LINKED ISSUE #208: Adding API_KEY dialog form accepts empty values
### Report

**Given**  I am user testing a prompt
**When** I click on the test button
**Then** I can insert an API_KEY for the test to work
**When** I click the "paste" button (pasting empty value)
**Then** I managed to go through the flow without getting an error that my input was invalid.

image

### Expected behavior

Inputs should have value validation.

### Steps to reproduce the problem

Same as described above.

### Pezzo version

v0.6.1

### How do you use Pezzo?

Pezzo Cloud

### PR DESCRIPTION
**Pull request by adityapuri75**
Fixes #208 
resolved form validation issue in api field

### CODE REVIEW

File: apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx [hunk 1]
```diff
@@ -65,15 +65,19 @@ export const ProviderApiKeyListItem = ({
...
-      organizationId: currentOrgId,
-    });
+    if (!editValue) {
+      messageApi.error("API key is required");
+    } else {
+      await updateKeyMutation.mutateAsync({
+        provider,
+        value: editValue,
+        organizationId: currentOrgId,
+      });
```
Comment (arielweinberger) [inline apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx:75]: Please don't use the `messageApi` for this. We have the ability to render errors under their input field as we do in other places (e.g. prompt creation modal, environment creation modal). Suits better.

**Review by arielweinberger — CHANGES_REQUESTED** [review]

**Comment by arielweinberger** [comment]
Thanks for putting the time @adityapuri75. I've left some feedback. Please make sure to run `npx nx format:write --all` at the root of the repo to format it before committing, this way CI will pass.

**Comment by adityapuri75** [comment]
Thanks for the feedback @arielweinberger, i have updated the branch as you suggested. i don’t have much experience with the fe but i have took reference from the other files and made changes Please Review.

File: apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx [hunk 2]
```diff
@@ -36,6 +37,7 @@ export const ProviderApiKeyListItem = ({
 }: Props) => {
   const [messageApi, contextHolder] = message.useMessage();
   const { currentOrgId } = useCurrentOrganization();
+  const [form] = Form.useForm<{ name: string }>();
```
Comment (arielweinberger) [inline apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx:40]:
```suggestion
  const [form] = Form.useForm<{ apiKey: string }>();
```

**Comment by adityapuri75** [comment]
Hello @arielweinberger thanks for the feedback on my commit, can i start working on issue  #238

**Comment by arielweinberger** [comment]
@adityapuri75 Please comment there so I can assign it to you. Thanks for this contribution!

**Review by arielweinberger — APPROVED** [review]

**Pull request merged**